install_cluster.py

```diff
@@ -0,0 +1,260 @@
+
+        config_dict['manager']['private_ip'] = node.private_ip
+        config_dict['manager']['public_ip'] = node.public_ip
+        _create_config_file(config_dict, node.name)
+
+
+def _prepare_manager_config_files(instances_dict,
+                                  rabbitmq_credentials,
+                                  load_balancer_ip):
+    logger.info('Preparing Manager config files')
+    ca_path = join(REMOTE_CERTS_DIR, 'ca.pem')
+    for node in instances_dict['manager']:
+        config_dict = get_dict_from_yaml(join(JUMP_HOST_DIR,
+                                              'manager_config.yaml'))
+        config_dict['manager']['hostname'] = node.name
+        config_dict['manager']['private_ip'] = node.private_ip
+        config_dict['manager']['public_ip'] = node.public_ip
+        config_dict['manager']['cloudify_license_path'] = \
+            join(REMOTE_INSTALL_CLUSTER_DIR, 'license.yaml')
+        config_dict['rabbitmq']['cluster_members'] = \
+            _get_rabbitmq_cluster_members(instances_dict['rabbitmq'])
+        config_dict['rabbitmq']['username'] = rabbitmq_credentials[0]
+        config_dict['rabbitmq']['password'] = rabbitmq_credentials[1]
+        config_dict['rabbitmq']['ca_path'] = ca_path
+        config_dict['postgresql_server']['cluster']['nodes'] = \
+            _get_postgresql_cluster_members(instances_dict['postgresql'])
+        config_dict['postgresql_server']['ca_path'] = ca_path
+        if load_balancer_ip:
+            config_dict['networks'] = {'load_balancer': load_balancer_ip}
+        _write_certs_to_config(config_dict, 'manager', node.name)
+        _create_config_file(config_dict, node.name)
+
+
+def _install_instances(instances_dict, rpm_download_link):
+    for instance_type in INSTANCES_TYPES:
+        logger.info('installing %s instances', instance_type)
+        for instance in instances_dict[instance_type]:
+            logger.info('Copying the %s directory to %s',
+                        LOCAL_INSTALL_CLUSTER_DIR, instance.name)
+            instance.put_dir(LOCAL_INSTALL_CLUSTER_DIR,
+                             REMOTE_INSTALL_CLUSTER_DIR,
+                             override=True)
+            logger.info('Installing Cloudify RPM on %s', instance.name)
+            instance.run_command('curl -o {0} {1}'.format(REMOTE_RPM_PATH,
+                                                          rpm_download_link))
+            instance.run_command(
+                'yum install -y {}'.format(REMOTE_RPM_PATH), use_sudo=True)
+
+            config_path = join(REMOTE_CONFIG_DIR,
+                               '{}_config.yaml'.format(instance.name))
+            instance.run_command(
+                'cp {0} /etc/cloudify/config.yaml'.format(config_path))
+
+            logger.info('Installing %s', instance.name)
+            instance.run_command(
+                'cfy_manager install --private-ip {0} --public-ip {1}'.format(
```

Comment:
It might be good to put the IPs in the config file (I thought I saw that you did?) and then not specify them on the command line? Looks a bit cleaner (to me), at least.

Reply:
You're absolutely right, I put it in both places because I remembered we had an issue where the agents weren't able to connect to the manager if the IPs weren't present in the config.yaml file or in the install command. In any case, I'll delete them from the command, and test to see if this issue still exists.